### ISSUE TITLE
detect-object-injection: the holder-property exemption survives a replacement (FN)

### ISSUE THREAD

**Issue by ofri-peretz** [ofri-peretz/eslint#998]
Raised by CodeRabbit on #997 and **confirmed by reproduction**, not just by reading.

#997 added an exemption for a null-prototype map held as a property of a holder object (`const flags = { bools: Object.create(null) }`). It resolves the holder, finds the property in the *initializer*, and exempts the indexed write.

The initializer says what the property was **created** as. It does not say what it **is** at the write.

## Reproduction

Run against the rule as it stands on `main`. All three are silent; B and C should report.

```ts
// A — CONTROL, correctly exempt
const flags = { bools: Object.create(null) };
export function f(req: any) { const k = req.query.k; flags.bools[k] = 1; }

// B — the property is replaced with an ordinary object
const flags = { bools: Object.create(null) };
flags.bools = {};
export function f(req: any) { const k = req.query.k; flags.bools[k] = 1; }

// C — a spread rebuild replaces it
let flags = { bools: Object.create(null) };
flags = { ...flags, bools: {} };
export function f(req: any) { const k = req.query.k; flags.bools[k] = 1; }
```

B and C write an attacker-controlled key into a **prototype-bearing** object and the rule stays silent. That is a false negative in the rule whose entire purpose is prototype pollution — the direction that matters most here.

## Why this is not a quick fix

I tried the conservative narrowing — withdraw the exemption if the holder binding or that property is written anywhere in the file — and it broke five previously-passing exempt cases, including the burgee one (`packages/burgee/src/yargs-parser.ts:188`) that motivated the exemption. Proving *which* value reaches the write needs flow analysis the rule does not currently have, so a correct fix is real work rather than a guard clause. Reverted rather than shipped half-done.

## Options

1. **Narrow the exemption properly** — track writes between the declaration and the indexed access. Correct, and the most work.
2. **Require the holder to be `const` and the property never assigned** — cheaper, still needs a write-scan that does not misfire on the five exempt cases.
3. **Withdraw the holder-property exemption** and take the false positives back. Recall-safe, and the repo's own stance is that "a recall loss is the worse trade for a security rule".

Recorded against a reproduction so the choice is made against evidence rather than an argument.
